packages/web3-core-types/src/types.ts

```diff
@@ -177,7 +177,8 @@ export interface Eth2RequestArguments extends RequestArguments {
 export interface RpcResponse {
     id: number;
     jsonrpc: string;
-    result: any;
+    result?: any;
+    error?: any;
```

Comment:
I consider its **JSON PRC 2** spec, in that case this type `RpcResponse` is not correct. Both of the `result` and `error` can't be optional the same time. Also when `error` is present it have a particular and fixed structure which implies it MUST contain `code` attribute.  See reference https://www.jsonrpc.org/specification#response_object

Reply:
and as per specs these cant be mandatory at same time:

result
This member is REQUIRED on success. >> required
This member MUST NOT exist if there was an error invoking the method. >> optional 
The value of this member is determined by the method invoked on the Server.
error
This member is REQUIRED on error. >> required
This member MUST NOT exist if there was no error triggered during invocation. >> optional
The value for this member MUST be an Object as defined in section 5.1.

Reply:
I am fully aware of JSON Rpc specs. Had implemented  it many times already. If you take a close look at specs you mentioned `result` and `error` can't be optional the same time. For any given RPC Response there MUST exists either the `result` or `error`. Its an important type so we have to fix it.